bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -179,13 +180,49 @@ private void refresh() {
         }
     }
 
+    /**
+     * Check if the data needs to be refreshed.
+     *
+     * The data is refreshed if:
+     * - the thing is offline
+     * - the local cache is empty
+     * - the current time is after 15:00 and the last refresh was more than an hour ago
+     * - the current time is after 18:00 and the last refresh was more than an hour ago
+     * - the current time is after 21:00 and the last refresh was more than an hour ago
+     *
+     * @return true if the data needs to be refreshed
+     */
     private boolean needRefresh() {
+        // if the thing is offline, we need to refresh
         if (getThing().getStatus() != ThingStatus.ONLINE) {
             return true;
         }
-        SortedSet<AwattarPrice> localPrices = prices;
-        return localPrices == null
-                || localPrices.last().timerange().start() < Instant.now().toEpochMilli() + 9 * 3600 * 1000;
+
+        // if the local cache is empty, we need to refresh
+        if (prices != null) {
```

Comment:
This must be `if (prices == null) {`.

I will open a fix pr with tests on this soon.

Reply:
For reference: https://github.com/openhab/openhab-addons/pull/17145